### PR TITLE
drivers: sensor: bmg160: Add multi-instance support

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
Move driver to use DT_INST_FOREACH_STATUS_OKAY to add
multi-instance support.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

File: drivers/sensor/bmg160/bmg160.c [hunk 1]
```diff
@@ -327,19 +330,28 @@ int bmg160_init(const struct device *dev)
...
-		    &bmg160_api);
+#define BMG160_DEFINE(inst)									\
+	struct bmg160_device_data bmg160_data_##inst;						\
+												\
+	static const struct bmg160_device_config bmg160_config_##inst = {			\
+		.i2c = I2C_DT_SPEC_INST_GET(inst),						\
+		IF_ENABLED(CONFIG_BMG160_TRIGGER,						\
+			   (COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, int_gpios),			\
+					(.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),),	\
+					())))							\
```
Comment (gmarull) [inline drivers/sensor/bmg160/bmg160.c:349]: `GPIO_DT_SPEC_INST_GET_OR()`

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/bmg160/bmg160.c (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/bmg160/bmg160.c]: Fixed!

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/bmg160/bmg160.c [hunk 2]
```diff
@@ -327,19 +330,26 @@ int bmg160_init(const struct device *dev)
...
-	IF_ENABLED(CONFIG_BMG160_TRIGGER,
-		   (.int_gpio = GPIO_DT_SPEC_INST_GET(0, int_gpios),))
-};
-
-DEVICE_DT_INST_DEFINE(0, bmg160_init, NULL,
-		    &bmg160_data,
-		    &bmg160_config, POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,
-		    &bmg160_api);
+#define BMG160_DEFINE(inst)									\
+	struct bmg160_device_data bmg160_data_##inst;						\
```
Comment (MaureenHelm) [inline drivers/sensor/bmg160/bmg160.c:342]:
```suggestion
	static struct bmg160_device_data bmg160_data_##inst;						\
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/bmg160/bmg160.c (hunk 2 above)
Comment (benjaminbjornsson) [reply drivers/sensor/bmg160/bmg160.c]: Fixed!

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**